lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2279,15 +2279,20 @@ public function getDefaultValueDeclarationSQL($field)
         if (isset($field['default'])) {
             $default = " DEFAULT '".$field['default']."'";
             if (isset($field['type'])) {
-                if (in_array((string) $field['type'], ["Integer", "BigInt", "SmallInt"])) {
+                $type = $field['type'];
+                if ($type instanceof Types\IntegerType ||
+                    $type instanceof Types\BigIntType ||
+                    $type instanceof Types\SmallIntType
```

Comment:
Should I introduce a parent class? This test is done several time ( it done in SQLServerPlatform, for instance)

Reply:
Or maybe a marker interface?

Reply:
Do you mean something like an AbstractNumberType? It wouldn't introduce number-specific functionality and just be an empty class, if you meant such a solution. A marker interface would have no methods, right? I'm not sure about new class or interface files, that are only introduced for checks. Do those have a purpose in userland for custom types or would those be only used internally?

I also considered (static?) methods for internal type checking, but those had to be maintained too and aren't elegant either. It's like we can only introduce more complexity to Doctrine.

Reply:
> Do you mean something like an AbstractNumberType? It wouldn't introduce number-specific functionality and just be an empty class, if you meant such a solution. 

Correct, but indeed, an interface is better

> A marker interface would have no methods, right?

yup

> I'm not sure about new class or interface files, that are only introduced for checks. Do those have a purpose in userland for custom types or would those be only used internally?

People that introduce numeric types and would like to benefit from the code behind those checks could implement that interface, so yeah, there is a purpose in userland.

> I also considered (static?) methods for internal type checking, but those had to be maintained too and aren't elegant either. It's like we can only introduce more complexity to Doctrine.

Yeah, and I think a marker interface is really not a big thing to make public.

Reply:
👍 for the marker interface. I'd suggest to mark it as internal to tell people to simply no use it (`@internal`).